commit e8c5a8f996040d1baa9568052341a7d18083b991
Author: Konstantin Kolinko <kkolinko@apache.org>
Date:   Tue Jan 5 03:17:36 2016 +0000

    Mark ReadWriteLock in modeler classes as final. (In review of r1722660/r1722661/r1722663 of Tomcat 7).
    
    The lock itself is accessed by multiple threads, thus it is better to mark such field as final to ensure consistency.
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1722996 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/tomcat/util/modeler/NotificationInfo.java b/java/org/apache/tomcat/util/modeler/NotificationInfo.java
index d3fde88..4929b83 100644
--- a/java/org/apache/tomcat/util/modeler/NotificationInfo.java
+++ b/java/org/apache/tomcat/util/modeler/NotificationInfo.java
@@ -41,7 +41,7 @@ public class NotificationInfo extends FeatureInfo {
      */
     transient MBeanNotificationInfo info = null;
     protected String notifTypes[] = new String[0];
-    protected ReadWriteLock notifTypesLock = new ReentrantReadWriteLock();
+    protected final ReadWriteLock notifTypesLock = new ReentrantReadWriteLock();
 
     // ------------------------------------------------------------- Properties
 
diff --git a/java/org/apache/tomcat/util/modeler/OperationInfo.java b/java/org/apache/tomcat/util/modeler/OperationInfo.java
index a5dc8bf..9197d64 100644
--- a/java/org/apache/tomcat/util/modeler/OperationInfo.java
+++ b/java/org/apache/tomcat/util/modeler/OperationInfo.java
@@ -48,7 +48,7 @@ public class OperationInfo extends FeatureInfo {
 
     protected String impact = "UNKNOWN";
     protected String role = "operation";
-    protected ReadWriteLock parametersLock = new ReentrantReadWriteLock();
+    protected final ReadWriteLock parametersLock = new ReentrantReadWriteLock();
     protected ParameterInfo parameters[] = new ParameterInfo[0];
 
 
